### PR TITLE
feat: preserve the sorting searching and pagination in dashboard page

### PR DESCRIPTION
**Pull request by keshavg2**
### Summary
#3938 

before 
https://www.loom.com/share/728ec188880043919e795f03049fc8a8?sid=75cea3c5-4b43-4245-b0c7-79a15410e977

after
https://www.loom.com/share/6cef88de183445ebb4f7ae9fc3b83a6d

#### Affected Areas and Manually Tested Areas
I'm checking the alerts page, service page, and exception page table

### CODE REVIEW

**Comment by keshavg2** [comment]
Hi @YounixM Please review this PR.

File: frontend/src/container/ListOfDashboard/utils.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export const filterDashboard = (
+	searchValue: string,
+	dashboardList: any,
```
Comment (Rajat-Dabade) [inline frontend/src/container/ListOfDashboard/utils.ts:3]: `any` can be replaced with `Dashboard[]` can be imported from 

`import { Dashboard } from 'types/api/dashboard/getAll';`

File: frontend/src/container/ListOfDashboard/utils.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+): any[] => {
+	// Convert the searchValue to lowercase for case-insensitive search
+	const searchValueLowerCase = searchValue.toLowerCase();
+
+	// Use the filter method to find matching objects
+	return dashboardList.filter((item: any) => {
```
Comment (Rajat-Dabade) [inline frontend/src/container/ListOfDashboard/utils.ts:9]: `any` can be replaced with `Dashboard`, as above.

File: frontend/src/container/ListOfDashboard/utils.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+		// Convert each property value to lowercase for case-insensitive search
+		const itemValues = Object.values(item?.data).map((value: any) =>
```
Comment (Rajat-Dabade) [inline frontend/src/container/ListOfDashboard/utils.ts:11]: Once the above `any` is replaced with `Dashboard`, no need to `(value: any)`, can be replaced by `(value)`, as the typescript will infer it internally.

**Review by Rajat-Dabade — APPROVED** [review]
LGTM 🚀  , Nice work @keshavg2 🎉

**Comment by keshavg2** [comment]
Thank you, @Rajat-Dabade, for your review of the PR. I've incorporated the suggested changes.

**Review by YounixM — APPROVED** [review]

**Comment by YounixM** [comment]
Approved. Thanks for contributing @keshavg2

**Pull request merged**